test/Microsoft.ML.Tests/TrainerEstimators/MatrixFactorizationTests.cs

```diff
@@ -53,6 +53,23 @@ public void MatrixFactorization_Estimator()
             Done();
         }
 
+        [Fact]
+        public void MatrixFactorization_Duplicate_Header()
+        {
+            string labelColumnName = "Label";
+            string matrixColumnIndexColumnName = "Col";
+            string matrixRowIndexColumnName = "Row";
+
+            var data = new TextLoader(Env, GetLoaderArgs(labelColumnName, matrixColumnIndexColumnName, matrixRowIndexColumnName))
+                    .Load(new MultiFileSource(GetDataPath(TestDatasets.trivialMatrixFactorization.trainFilename)));
+
+            var dataDuplicateHeader = new TextLoader(Env, GetLoaderArgs(labelColumnName, matrixColumnIndexColumnName, matrixRowIndexColumnName))
+                    .Load(new MultiFileSource(GetDataPath(TestDatasets.trivialMatrixFactorizationWithDuplicateHeader.trainFilename)));
```

Comment:
Calling `.Load()` won't actually load the data, so I'm not sure: is this test actually hitting the code you've introduced in the `LineReader()`?

To make sure those new lines are being tested, I'd think it's necessary to actually open cursors to go through both `data` and `dataDuplicateHeader`.

Also, please move this test to `TextLoaderTests` inside `TextLoaderTests.cs` because this is a TextLoader test, not a MF test. Thanks.

Reply:
CheckSameValues method will iterate both data view and read actual data from text file and do a comparison.

I will move this test to TextLoaderTests, thanks.

---
In reply to: [448682192](https://github.com/dotnet/machinelearning/pull/5275#discussion_r448682192) [](ancestors = 448682192)

Reply:
Great, thanks.

---
In reply to: [448687068](https://github.com/dotnet/machinelearning/pull/5275#discussion_r448687068) [](ancestors = 448687068,448682192)